objects/src/testing/account_code.rs

```diff
@@ -30,9 +30,9 @@ pub const DEFAULT_AUTH_SCRIPT: &str = "
 ";
 
 impl AccountCode {
-    /// Creates a mock [AccountCode] that exposes wallet interface
-    pub fn mock_wallet(assembler: Assembler) -> AccountCode {
-        AccountCode::new(Self::mock_library(assembler)).unwrap()
+    /// Creates a mock [AccountCode]
+    pub fn mock_account_code(assembler: Assembler, is_faucet: bool) -> AccountCode {
+        AccountCode::new(Self::mock_library(assembler), is_faucet).unwrap()
     }
```

Comment:
Instead of doing it like this, I would probably keep `mock_wallet()` method and add `mock_faucet()` method.

Reply:
I changed the name from `mock_wallet` to `mock_account_code` because the provided account code is not the one of a wallet anymore. It's a mix of both `faucet`, `wallet`, `utility procedures` and `dummy procedures`:

https://github.com/0xPolygonMiden/miden-base/blob/76a6635593ca168eda4517061ee0b4b845ad1c0f/objects/src/testing/account_code.rs#L38-L126

Reply:
Given that regular accounts behave different from faucets - I wonder if it make sense to separate them. We'd basically have 2 methods: `mock_account_code()` and `mock_faucet_code()`. The former would not have any faucet-related methods.

Reply:
Would make sense to separate them imo. 

Could be part of another PR considering that it would be a refactor touching 10+ files.

Reply:
Opened an issue here: https://github.com/0xPolygonMiden/miden-base/issues/878